File: 24-9/ApiClient.java
package com.example.retrofitex;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiClient
{
    static Retrofit retrofit=null;

    public static final String BASE_URL="https://vyasprakruti.000webhostapp.com/InventorymanaementSystem/";
    public static Retrofit getapiclient()
    {

        retrofit= new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        return retrofit;
    }
}

File: 10-11/Project/Client/ApiClient.kt
package com.example.e_greetingsheet.Client

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class ApiClient {

    companion object {

        var url = "https://vyasprakruti.000webhostapp.com/E-Greetings%20Project/"
        fun getapiclient(): Retrofit {
            var retrofit: Retrofit? = null

            if (retrofit == null) {
                retrofit = Retrofit.Builder()
                    .baseUrl(url)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
            }


            return retrofit!!
        }

    }
}

